logger_test.go

```diff
@@ -136,10 +151,19 @@ func TestJSONLoggerLog(t *testing.T) {
 		jl.Log(Debug, "foo")
 		assertMessage(t, "debug", "foo", output()[0])
 	})
+
 	withJSONLogger(t, nil, func(jl *jsonLogger, output func() []string) {
 		assert.Panics(t, func() { jl.Log(Panic, "foo") }, "Expected logging at Panic level to panic.")
 		assertMessage(t, "panic", "foo", output()[0])
 	})
+
+	stub := stubExit()
+	defer stub.Unstub()
+	withJSONLogger(t, nil, func(jl *jsonLogger, output func() []string) {
+		jl.Log(Fatal, "foo")
+		assertMessage(t, "fatal", "foo", output()[0])
+		assert.Equal(t, 1, *stub.Status, "Expected to call os.Exit with status 1.")
```

Comment:
optional nit: if `exit` isn't called, rather than failing with a nice error, it'll cause a panic. we could do
```go
if assert.NotNil(t, stub.Status, "Expected a call to os.Exit") {
  assert.Equal(t, 1, *stub.Status, "os.Exit got unexpected exit code")
}
```

(i also try to not mention the specific value we expect in the message, since `Equal` will print the value it expected vs what it got already, and if we change it, it means less places to change / accidentally miss)